src/Kestrel.Core/ListenOptions.cs

```diff
@@ -17,7 +17,9 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Core
     /// Describes either an <see cref="IPEndPoint"/>, Unix domain socket path, or a file descriptor for an already open
     /// socket that Kestrel should bind to or open.
     /// </summary>
+#pragma warning disable PUB0001 // Pubternal type in public API
     public class ListenOptions : IEndPointInformation, IConnectionBuilder
+#pragma warning restore PUB0001 // Pubternal type in public API
```

Comment:
It should be allowed to implement a pubternal interface like you can with an internal one.